### DISCUSSION TITLE
Missing information for installing server 3.8.0 Linux

### DISCUSSION THREAD

**Discussion by gene96817** [jamulussoftware/jamulus#1852] [Documentation]
I started to upgrade my server from 3.7.0 to 3.8.0. I am wishing the documentation was explicit about the following.
Are they in a place I overlooked?

- Creating the jamulus_welcome file. Please include an example. Where is the recommended location for the file?
- Command-Line Options
    - these commands are not mentioned. Are they no longer used or supported? 
          -s     -w        -u        -T     and  --serverpublicip
    - does --serverbindip replace --serverpublicip?   The explanation is too terse for me to understand.

The page "Running a Server" has the first heading "Installation for Windows and macOS". Yet the text below has a third section for Linux. 

It would be good to have some text on recommended steps for upgrading from 3.7.0 to 3.8.0.
-- Can we upgrade without removing 3.7.0? (I hope so.)
-- What is the best way to save the 3.7.0 installation just in case it is necessary to rollback from a 3.8.0 installation?
-- Can we use the jamulus.service and jamulus_welcome files without change?
------- Are there elements that would be fatal to 3.8.0?
------- Are there elements that must be added for 3.8.0 to operate properly? 

Thanks....

**Comment by npostavs** [comment]
> * Creating the jamulus_welcome file. Please include an example. Where is the recommended location for the file?

I'm confused why you need a recommended location. You put it anywhere you like.

**Reply by gene96817** [reply]
Sure it can go anywhere. I can handle that.
However, if there is a recommended (arbitrary) place, the consistency could make it easier to support.

**Reply by npostavs** [reply]
Wouldn't it be easier not to recommend a welcome file at all?

**Reply by gene96817** [reply]
Since the welcome message is very useful, we should have some documentation and example. Our developer community is very facile with these technologies. We need to remember to make it easy for other people to adopt Jamulus.

**Reply by henkdegroot** [reply]
I believe some work is being done to write a Servers Administrator Manual....so hoping that will be included

**Comment by henkdegroot** [comment]
> * Command-Line Options

According to the help (Jamulus --help), these options still exists. So I agree they seem to be missing from the webpage now. There is already an issue reported for that to fix it. See [#520](https://github.com/jamulussoftware/jamuluswebsite/issues/520)

**Comment by henkdegroot** [comment]
> The page "Running a Server" has the first heading "Installation for Windows and macOS". Yet the text below has a third section for Linux.

Are you sure "see" the latest website/pages? Perhaps something is in cache or even a different website?

When I go to this url [https://jamulus.io/wiki/Running-a-Server](https://jamulus.io/wiki/Running-a-Server), there is first text to explaining and then the below is shown:

![server-page](https://user-images.githubusercontent.com/13550012/121304083-2e5b9600-c8fc-11eb-9229-61ee5ddf121d.PNG)

Just two buttons, first for Windows / Mac and the second for Linux.

**Reply by hsweet** [reply]
I installed 3.8 headless from the  .deb file on an Ubuntu server this week. The confusing part is that after that jamulus -v still showed 2.7.  Turned out I had to change the executable in the .service file from /usr/bin/jamulus to /usr/bin/jamulus-headless, restart the service and now I am running 3.8 but can go back in a minute.
The other confusing part on Ubuntu is where stuff ends up doing an install from source vs from a .deb file

**Reply by rdica** [reply]
If you're compiling anything from source and readme or other says `make install`, you can `make -n install` (its a dry run) and that will show where everything will get installed. Src installs tend to use `/usr/local` as the prefix for destination directories by default unless otherwise specified.

**Comment by chrisrimple** [comment]
Related, I did a clean/new install of 3.8.0 with commands:

```
cd jamulus
git checkout latest
qmake "CONFIG+=nosound headless" Jamulus.pro
make clean
make
make install
```

As far as I can tell, no default jamulus.service file is created in this process, but I think that it should be.  OR an example file should be available somewhere.  Because I had to create it manually, following instructions that I found on a non-Jamulus website, and which likely have out-of-date contents for the file.

**Reply by npostavs** [reply]
There's an example in `distributions/jamulus-server.service`. I  guess `distributions/` refers to "Linux distributions", but that is not very obvious.

**Reply by rdica** [reply]
Found this: https://github.com/jamulussoftware/jamulus/blob/master/distributions/jamulus-server.service

```[Unit]
Description=Jamulus-Server
After=network.target
StartLimitIntervalSec=0

[Service]
Type=simple
Restart=always
RestartSec=1
User=jamulus
ExecStart=/bin/sh -c 'exec /usr/bin/jamulus -s -n -l /var/log/jamulus -e anygenre3.jamulus.io -g -o "$(uname -n);;"'

[Install]
WantedBy=multi-user.target
```
The same file is in `distributions/` subdirectory of the directory you built in.
Hope that helps.

**Reply by henkdegroot** [reply]
And there is a "headless" version in the `distributions/debian` folder as well.
